File: leetcode-submissions/1400_Construct_K_Palindrome_Strings.cpp
class Solution {
public:
    bool canConstruct(string s, int k) {
        if(s.size()<k){
            return false;
        }
        unordered_map<char,int>mp;
        for(int i=0;i<s.size();i++){
            mp[s[i]]++;
        }
        int oddCount =0;
        for(auto ele :mp){
if(ele.second%2){
    oddCount++;
}
        }
        return oddCount<=k;
    }
};

File: leetcode-submissions/1400_Construct_K_Palindrome_Strings.ts
function canConstruct(s: string, k: number): boolean {
    if(s.length<k){
        return false;
    }
    const map:Map<string,number>=new Map();
    for(let i=0;i<s.length;i++){
        if(map.has(s[i])){
            map.set(s[i],map.get(s[i])+1);
        }else{
            map.set(s[i],1);
        }
    }
    let oddcount :number =0;
    for(const [key ,value] of map.entries()){
if(value%2===1){
    oddcount++;
}
    }
    return oddcount<=k;
};